saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticatedPrincipal.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.provider.service.authentication;
+
+import java.io.Serializable;
+
+import org.springframework.security.core.AuthenticatedPrincipal;
+
+public interface Saml2AuthenticatedPrincipal extends AuthenticatedPrincipal, Serializable {
```

Comment:
Could we make `SimpleSaml2AuthenticatedPrincipal` implement `Serializable` instead? 
Then it will be consistent with `OAuth2AuthenticatedPrincipal` and `DefaultOAuth2AuthenticatedPrincipal`.

Reply:
You're right. Done.